packages/twenty-server/src/engine/core-modules/code-engine/utils/build-directory-manager.ts

```diff
@@ -0,0 +1,45 @@
+import { join } from 'path';
+import { tmpdir } from 'os';
+import fs from 'fs';
+
+import fsExtra from 'fs-extra';
+import { v4 } from 'uuid';
+
+const TEMPORARY_LAMBDA_FOLDER = 'twenty-build-lambda-temp-folder';
+const TEMPORARY_LAMBDA_SOURCE_FOLDER = 'src';
+const LAMBDA_ZIP_FILE_NAME = 'lambda.zip';
+const LAMBDA_ENTRY_FILE_NAME = 'index.js';
+
+export class BuildDirectoryManager {
+  private temporaryDir = join(tmpdir(), `${TEMPORARY_LAMBDA_FOLDER}_${v4()}`);
+  private lambdaHandler = `${LAMBDA_ENTRY_FILE_NAME.split('.')[0]}.handler`;
+
+  async init() {
+    const sourceTemporaryDir = join(
+      this.temporaryDir,
+      TEMPORARY_LAMBDA_SOURCE_FOLDER,
+    );
+    const lambdaZipPath = join(this.temporaryDir, LAMBDA_ZIP_FILE_NAME);
+    const javascriptFilePath = join(sourceTemporaryDir, LAMBDA_ENTRY_FILE_NAME);
+
+    if (!fs.existsSync(this.temporaryDir)) {
+      fs.mkdirSync(this.temporaryDir);
+      fs.mkdirSync(sourceTemporaryDir);
+    } else {
+      await fsExtra.emptyDir(this.temporaryDir);
+      fs.mkdirSync(sourceTemporaryDir);
+    }
+
+    return {
+      sourceTemporaryDir,
+      lambdaZipPath,
+      javascriptFilePath,
+      lambdaHandler: this.lambdaHandler,
+    };
+  }
+
+  async clean() {
+    await fsExtra.emptyDir(this.temporaryDir);
+    fs.rmdirSync(this.temporaryDir);
```

Comment:
Same question as above. What is the reason behind mixing sync and async I/O operations? (fsExtra's emptyDir function is async)